### PR TITLE
Minor improvements for the finance module

### PR DESCRIPTION
**Pull request by ericnordelo**

### CODE REVIEW

File: docs/modules/ROOT/pages/api/finance.adoc [hunk 1]
```diff
@@ -208,11 +208,13 @@ Returns the total vested amount of a specified `token` at a given `timestamp`.
 
 Releases the amount of a given `token` that has already vested and returns that amount.
 
+NOTE: If the releasable amount is zero, this function won't emit the event.
+
```
Comment (andrew-fleming) [inline docs/modules/ROOT/pages/api/finance.adoc:212]: Should we mention that `ERC20::transfer` won't be attempted when amt == 0 as well?

File: packages/finance/src/vesting/vesting.cairo [hunk 2]
```diff
@@ -132,21 +136,25 @@ pub mod VestingComponent {
...
+            if amount > 0 {
+                self.Vesting_released.write(token, self.Vesting_released.read(token) + amount);
+                self.emit(AmountReleased { token, amount });
+
+                let beneficiary = get_dep_component!(@self, Ownable).owner();
+                let token_dispatcher = IERC20Dispatcher { contract_address: token };
+                assert(
+                    token_dispatcher.transfer(beneficiary, amount), Errors::TOKEN_TRANSFER_FAILED
+                );
+            }
```
Comment (andrew-fleming) [inline packages/finance/src/vesting/vesting.cairo:157]: I think we should add a test for events not emitting when amount == 0

File: docs/modules/ROOT/pages/api/finance.adoc [hunk 3]
```diff
@@ -91,7 +91,7 @@ Returns the total vested amount of a specified `token` at a given `timestamp`.
 
 Releases the amount of a given `token` that has already vested and returns that amount.
 
-Emits {AmountReleased} event.
+Emits an {AmountReleased} event.
```
Comment (andrew-fleming) [inline docs/modules/ROOT/pages/api/finance.adoc:94]: Should we amend this to "may emit" as well? It appears to be an unfaithful implementation if the interface says "emits" while the impl of the interface says "may emit"

**Review by andrew-fleming — COMMENTED** [review]
Good call on the proposed change! I left a few comments

File: docs/modules/ROOT/pages/api/finance.adoc [hunk 4]
```diff
@@ -208,11 +208,13 @@ Returns the total vested amount of a specified `token` at a given `timestamp`.
...
 Releases the amount of a given `token` that has already vested and returns that amount.
 
+NOTE: If the releasable amount is zero, this function won't emit the event.
+
 Requirements:
 
 - `transfer` call to the `token` must return `true` indicating a successful transfer.
 
-Emits {AmountReleased} event.
+May emit an {AmountReleased} event.
```
Comment (immrsd) [inline docs/modules/ROOT/pages/api/finance.adoc:217]: Shall we specify here when an event won't be emitted? It feels bit weird given no context

**Review by immrsd — APPROVED** [review]
LGTM

File: docs/modules/ROOT/pages/api/finance.adoc (hunk 4 above)
Comment (ericnordelo) [reply docs/modules/ROOT/pages/api/finance.adoc:218]: Do you think the NOTE in the description is not enough?

**Review by andrew-fleming — APPROVED** [review]
LGTM!

**Pull request merged**